drivers/spi/spi_mcux_lpspi.c

```diff
@@ -325,23 +325,60 @@ static void spi_mcux_dma_callback(const struct device *dev, void *arg,
 	if (data->ctx.asynchronous &&
 	((data->status_flags & SPI_MCUX_LPSPI_DMA_DONE_FLAG)  ==
 	SPI_MCUX_LPSPI_DMA_DONE_FLAG)) {
-		/* Load dma blocks of equal length */
-		size_t dma_size = MIN(data->ctx.tx_len, data->ctx.rx_len);
-
-		if (dma_size == 0) {
-			dma_size = MAX(data->ctx.tx_len, data->ctx.rx_len);
-		}
+		/* dma_size of completed transfers were the same for TX and RX */
+		size_t dma_size = data->ctx.tx_len;
 
 		spi_context_update_tx(&data->ctx, 1, dma_size);
 		spi_context_update_rx(&data->ctx, 1, dma_size);
 
 		if (data->ctx.tx_len == 0 && data->ctx.rx_len == 0) {
 			spi_context_complete(&data->ctx, spi_dev, 0);
+		} else {
+			/* Restart DMA for next buffer */
+			/* Load dma blocks of equal length */
+			dma_size = MIN(data->ctx.tx_len, data->ctx.rx_len);
+
+			if (dma_size == 0) {
+				dma_size = MAX(data->ctx.tx_len, data->ctx.rx_len);
```

Comment:
why is it not always max?

Reply:
Hi @decsny ,
Making this change (always MAX) here in `spi_mcux_dma_callback()`, the `spi_loopback` test passes.  However, if I make this same change in `spi_mcux_dma_rxtx_load()`, the "half start" test fails.  I believe this existing code is keeping the RX and TX transfers equal size until one completes, and then it finishes the other.